projects/kyber/index.js

```diff
@@ -100,26 +101,50 @@ query get_tvl($block: Int) {
       }
     });
     const {dmmFactory} = await request(
-      graphUrl,
-      graphQuery,
+      egraphUrl,
+      egraphQuery,
       {
         block,
       }
+      
     );
     if(dmmFactory !== null){ // Has been created
-      const dmmTvlInUsdt = (Number(dmmFactory.totalLiquidityUSD)* 1e6).toFixed(0)
-      sdk.util.sumSingleBalance(balances, usdtAddress, dmmTvlInUsdt)
+      const ethTVL = (Number(dmmFactory.totalLiquidityUSD)* 1e6).toFixed(0)
+      sdk.util.sumSingleBalance(balances, usdtAddress, ethTVL)
     }
 
     return balances;
   }
+  async function polygonTvl(){
+    var polygonEndpoint ='https://api.thegraph.com/subgraphs/name/piavgh/dmm-exchange-matic';
+      var query = gql`
+      {
+        dmmFactories(first: 1) {
```

Comment:
I would prefer if you could pin this to the block number being passed as a parameter. There's two reasons for this:
- It allows us to get historical data correctly
- On polygon and other evm chains with high throughput it's common that applications that see a lot of traffic end up with outdated subgraphs (speed of updates is higher than the speed at which the subgraph can process them, so it stays constantly outdated). By pinning responses to a blockheight we make sure that, if the subgraph hasn't catched up, the response fails, thus preventing any misreporting.

Regarding the 2nd point I also wanted to ask if your polygon subgraph is just a copy of the ethereum one or it has been made lighter, as otherwise it's likely that this will happen and this code will need to be replaced with something else that reads data from onchain calls.

Reply:
Sure, to include block number makes sense.
And yes, it's copy of ethereum.

Reply:
changes done. Please review and help merge